pennylane/transforms/condition.py

```diff
@@ -60,11 +61,23 @@ def __init__(
 
 
 def cond(condition, true_fn, false_fn=None):
-    """Condition a quantum operation on the results of mid-circuit qubit measurements.
+    """A :func:`~.qjit` compatible decorator for if-else conditionals in PennyLane/Catalyst.
 
-    Support for using :func:`~.cond` is device-dependent. If a device doesn't
-    support mid-circuit measurements natively, then the QNode will apply the
-    :func:`defer_measurements` transform.
+    This can be used to condition a quantum operation on the results of mid-circuit qubit measurements
+    with the Python interpreter while acts like the traditional if-else conditional when is called inside a
+    QJIT decorated workflow. This means that each execution path, 'if' and 'else' branchs, is provided
+    as a separate function. All functions will be traced during compilation, but only one of them will
+    be executed at runtime, depending on the value of one or more Boolean predicates. The JAX equivalent
+    is the ``jax.lax.cond`` function, but this version is optimized to work with quantum programs in PennyLane.
+    This version also supports an 'else if' construct which the JAX version does not. However to use `else if`
+    you need to use `catalyst.cond <https://docs.pennylane.ai/projects/catalyst/en/latest/code/api/catalyst.cond.html>`__.

```

Comment:
@maliasadi I think we can remove this paragraph, since the paragraph and examples below explain this behaviour 🤔

Reply:
Otherwise, we can shorten it:
```suggestion
    When used with the :func:`~.qjit` decorator with a hybrid
    quantum-classical compiler, this function allows for general
    if-else constructs. Both the ``true_fn`` and ``false_fn`` branches
    will be captured by the compiler, with the executed branch determined
    at runtime (similar to ``jax.lax.cond``). For more details, as well
    as ``elseif`` support, please see :func:`catalyst.cond`.
    
    Without the :func:`~.qjit` decorator (**interpreted mode**), it is
    restricted to simply branching on mid-circuit measurement results.
```

Reply:
Nice! Thanks :+1: